Updates definition of `install_sources` (#778)
MicrosoftEdge/MSEdgeExplainers: WebInstall/explainer_cross_domain.md

}
```
####  **Install Sources manifest field**

The default behaviour of the cross-origin Web Install API can be to allow installations from any origin or from no origin. *This default is defined by the implementer*. The `install_sources` field in the manifest file specifies detailed origin installation permission allowing finer control by the web app on which origin(s) can install it.

> * **Cross-origin installations ON by default**: If the default behaviour is that an origin can be installed from any origin, then any origin can install an app from a different origin. In the presence of an `installed_sources` field, then the default is inverted and only the subset of origins specified in its value can install the app.

Reviewer: ```suggestion
* **Cross-origin installations ON by default**: If the default behaviour is that an origin can be installed from any origin, then any origin can install an app from a different origin. In the presence of an `install_sources` field, then the default is inverted and only the subset of origins specified in its value can install the app.
```

* **Cross-origin installations OFF by default**: If the implementer decides to make the default behaviour of cross-origin installations OFF by default, then no third party origin can install the web application. This is the most restrictive case, and the web app conforms to its usual behaviour of only being able to be installed from its same origin. If there is an `install_sources` field present in the manifest file, then only the specific set of origins specified in that value can install the app.

> In both cases, the `install_sources` field overrides the default behaviour adopted by the UA and allows only the specified origins to installed the application.

Reviewer: ```suggestion
In both cases, the `install_sources` field overrides the default behaviour adopted by the UA and allows only the specified origins to install the application.
```

}
```

##### Controlling installations from any origin
To ensure that the developer can have control of where their app can be installed from independent of the implementors default behaviour, a new web-manifest boolean key `allow_all_install_sources` can tell the UA that the application can be installed from any or no other origin.

* if set to `true`, the `install_sources` list is ignored (if included) and the app can be installed from any origin.

> * if set to `false`, the `install_sources` list is ignored (if included) and the app can be installed only from the same-origin it is hosted.

Reviewer: What about the case where the browser allows install by default, but the developer only wants to allow a few vetted stores to install? This would block that scenario.

What about:
```suggestion
* if set to `false`, the `install_sources` list is respected (if included) and the app can be installed only from the origins listed in `install_sources` or the same-origin it is hosted.
```

Reviewer: Chatted offline. Diego's suggestion was to change this to discuss the order of presidence: 
1. `install_sources` (if provided the UA assumes the dev wants to block by default)
2. `allow_all_install_sources`
3. UA default behavior of allow or block by default.

```json
{
    "name": "Awesome PWA",
    "display": "standalone",
    "start_url": "/index.html",

>     "all_all_install_sources": "true"

Reviewer 2: ```suggestion
    "allow_all_install_sources": "true"
```
